### PR TITLE
Improve 2d canvas

### PR DESCRIPTION
**Pull request by yustarandomname**
Improved Julia's draggables code to be more modular and reactive to the reset button

- Also added snapping functionality to both square and hexagonal grid

### CODE REVIEW

File: src/lib/p5-components/components/Canvas1.svelte
Comment (juliavdkris) [inline src/lib/p5-components/components/Canvas1.svelte:1]: This file can be removed

File: src/lib/p5-components/components/Vector.svelte [hunk 1]
```diff
@@ -21,31 +21,31 @@
...
-  }
+  // $: {
+  //   if (draggable && false) {
+  //     let draggablePos =
+  //       $draggables.get(key) || origin.clone().add(direction.clone().multiplyScalar(length)); // TODO: fix undefined thingy here
+  //     let v = draggablePos.clone().sub(origin);
+  //     direction = v.clone().normalize();
+  //     length = v.length();
+  //   }
+  // }
```
Comment (juliavdkris) [inline src/lib/p5-components/components/Vector.svelte:36]: Some commented out stuff here from my old implementation that I think can be removed

Comment (yustarandomname) [reply src/lib/p5-components/components/Vector.svelte]: Is removed in the latest version

**Review by yustarandomname — COMMENTED** [review]
The AbstractCanvas breaks all 3D images

**Review by PHengst — APPROVED** [review]
embed-r2r3 seems to be acting up, the rest of the applets and the code look good!

**Pull request closed without merging**